yew-macro/src/html_tree/mod.rs

```diff
@@ -82,6 +89,7 @@ impl ToTokens for HtmlTree {
             HtmlTree::Element(tag) => tag.to_tokens(tokens),
             HtmlTree::List(list) => list.to_tokens(tokens),
             HtmlTree::Block(block) => block.to_tokens(tokens),
+            HtmlTree::If(block) => block.to_tokens(tokens),
```

Comment:
I wasn't sure because it wasn't used. How can I trigger this?

Reply:
```rust
// define a component that wraps every child it receives in <li>

#[derive(Properties)]
struct Props {
  children: Children,
}

struct Comp { props: Props }
impl Component for Comp {
  type Properties = Props;

  fn view(&self) -> Html {
    // wrap every child in <li>
    html! {
      <ul>
        { for self.props.children.iter().map(|child| html! { <li>{ child }</li> }) }
      </ul>
    }
  }
}

// now let's test it

html! {
  <Comp>
    if true {
      <span>{ "hello" }</span>
      <span>{ "world" }</span>
    } else {
      <span>{ "goodbye" }</span>
      <span>{ "world" }</span>
    }
  </Comp>
}
```

This should always produce:
```html
<ul>
  <li><span>Hello</span></li>
  <li><span>World</span></li>
</ul>
```

`ToNodeIterator` is responsible for doing this. Its job is to create an iterator which yields the children individually instead of as a virtual list.

I think I already created an integration test for this somewhere, you can just add the `HtmlIf` there and see if it works (it won't work as it is right now).

Reply:
Ok I tried now. I even copied your example... but it didn't break. I forgot a bit the context to be honest so I will have to re-dig a bit. If you have an idea why it works (while it shouldn't) please let met know :sweat_smile: Here is the unexpectedly working commit 75b1a85c

Reply:
Ah but the CI won't run until I fix the conflicts XD

Reply:
Updated the branch. CI is passing.

I moved the test in order to use the headless browser: 83bcf239

Test is still passing for some reason.

Reply:
So this is what I have: `<ul><li><span>hello</span><span>world</span></li></ul>`

This is what you expected: `<ul><li><span>hello</span></li><li><span>world</span></li></ul>`

The <li> is not closing like you would expect but.... that's normal? Or am I missing something?

Reply:
> The `<li>` is not closing like you would expect but.... that's normal? Or am I missing something?

Right, currently `<li>` wraps the entire result of the if block, but I would expect it to wrap each `<span>` separately.

One reason why I think that should be the case is that I think that:
```rust
html! {
  <Comp>
    if true {
      <span>{ "hello" }</span>
      <span>{ "world" }</span>
    }
  </Comp>
}
```
should yield the same output as
```rust
html! {
  <Comp>
      <span>{ "hello" }</span>
      <span>{ "world" }</span>
  </Comp>
}
```


The other reason is that there simply isn't another way to do this right now. Yew doesn't have some kind of "unpack all children" syntax, but it's always possible to get the current result by wrapping the contents of the `if` block in fragments.

In most cases the user won't even notice it anyway, this is only an issue when a component inspects its children.